namespaces/nsenter/nsenter.c

```diff
@@ -151,6 +151,17 @@ void nsenter()
 			exit(1);
 		}
 	}
+	// blocking until the parent placed the process inside correct cgroups.
+	unsigned char s;
+	ssize_t ret;
+	do {
+		ret = read(3, &s, 1);
+	} while (ret == -1 && errno == EINTR);
```

Comment:
Thanks for the pointer @avagin . I had a lingering suspicion that it might not be required in this case.